### PR TITLE
Added count method.

### PR DESCRIPTION
**Pull request by darrenfang**
Added count method.

### CODE REVIEW

File: src/Dommel/DommelMapper.cs [hunk 1]
```diff
@@ -868,6 +869,57 @@ private static string BuildSelectSql<TEntity>(Expression<Func<TEntity, bool>> pr
         }
 
         /// <summary>
+        /// Count the entities matching the specified predicate.
```
Comment (henkmollema) [inline src/Dommel/DommelMapper.cs]:
```
Returns the number of entities matching the specified predicate.
```

File: src/Dommel/DommelMapper.cs [hunk 2]
```diff
@@ -868,6 +869,57 @@ private static string BuildSelectSql<TEntity>(Expression<Func<TEntity, bool>> pr
...
+        /// </returns>
+        public static long Count<TEntity>(this IDbConnection connection, Expression<Func<TEntity, bool>> predicate)
+        {
+            DynamicParameters parameters;
+            var sql = BuildCountSql(predicate, out parameters);
+            return connection.ExecuteScalar<long>(sql, parameters);
+        }
+
+        /// <summary>
+        /// Selects all the entities matching the specified predicate.
```
Comment (henkmollema) [inline src/Dommel/DommelMapper.cs]:
```
Returns the number of entities matching the specified predicate.
```

File: src/Dommel/DommelMapper.cs [hunk 3]
```diff
@@ -868,6 +869,57 @@ private static string BuildSelectSql<TEntity>(Expression<Func<TEntity, bool>> pr
         }
 
         /// <summary>
+        /// Count the entities matching the specified predicate.
+        /// </summary>
+        /// <typeparam name="TEntity">The type of the entity.</typeparam>
+        /// <param name="connection">The connection to the database. This can either be open or closed.</param>
+        /// <param name="predicate">A predicate to filter the results.</param>
+        /// <returns>
+        /// Count of entities of type <typeparamref name="TEntity"/> matching the specified
```
Comment (henkmollema) [inline src/Dommel/DommelMapper.cs]:
```
The number of entities matching the specified predicate.
```

File: src/Dommel/DommelMapper.cs [hunk 4]
```diff
@@ -868,6 +869,57 @@ private static string BuildSelectSql<TEntity>(Expression<Func<TEntity, bool>> pr
...
+        }
+
+        /// <summary>
+        /// Selects all the entities matching the specified predicate.
+        /// </summary>
+        /// <typeparam name="TEntity">The type of the entity.</typeparam>
+        /// <param name="connection">The connection to the database. This can either be open or closed.</param>
+        /// <param name="predicate">A predicate to filter the results.</param>
+        /// <returns>
+        /// A collection of entities of type <typeparamref name="TEntity"/> matching the specified
```
Comment (henkmollema) [inline src/Dommel/DommelMapper.cs]:
```
The number of entities matching the specified predicate.
```

**Comment by darrenfang** [comment]
Thank you.

**Comment by henkmollema** [comment]
I'll incorporate this in v2.

**Comment by henkmollema** [comment]
Merged in: d8afcc4e6a65c4ef275005a2f765f1a2a0c11e02.

**Pull request closed without merging**